File: Data Structures & Algorithms/kth-distinct-string-in-an-array/submission-0.java
class Solution {
    public String kthDistinct(String[] arr, int k) {
        Map<String, Integer> map = new HashMap();
        for (String s : arr) {
            map.merge(s, 1, (oldVal, newVal) -> oldVal + 1);
        }

        for (String s : arr) {
            if (map.get(s) == 1) {
                k--;
                if (k == 0) {
                    return s;
                }
            }
        }
        return "";
    }
}

File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        map = {}

        for c in s:
            if c not in map:
                map[c] = 1
            else:
                map[c] += 1
        
        for c in t:
            if c not in map:
                return False
            else:
                map[c] -= 1

        for val in map.values():
            if val != 0:
                return False 

        return True